jax/_src/dispatch.py

```diff
@@ -105,7 +105,7 @@ def apply_primitive(prim, *args, **params):
 RuntimeToken = Any
 
 class RuntimeTokenSet(threading.local):
-  tokens: Dict[core.Effect, Tuple[RuntimeToken, Device]]
+  tokens: dict[core.Effect, tuple[RuntimeToken, Device]]
```

Comment:
Subscriptable types requires Python 3.9, if I'm not mistaken. We still require compatibility with Python 3.7

Reply:
[See below](https://github.com/google/jax/pull/10678#discussion_r871665062)